### PR TITLE
[FIX] Change recording entity to REQUIRED for pet/blood modality

### PR DESCRIPTION
**Pull request by ghisvail**
To make it consistent with the [rules](https://github.com/bids-standard/bids-specification/blob/e47283a268f20b98d5c05ad1bb8c63da3d2a1ceb/src/schema/rules/datatypes/pet.yaml#L30) and the [examples](https://github.com/bids-standard/bids-examples/tree/master/pet003/sub-01/ses-01/pet)

### CODE REVIEW

**Comment by ghisvail** [comment]
Not sure what the circleci thing is about.

**Comment by mnoergaard** [comment]
Hi @ghisvail! Thanks for raising this issue. It states in the PET-BIDS spec that the recording entity is optional, unless several recordings are used (i.e. manual and autosampler).  So the example case you are referring to is not incorrect, but the recording entity could in principle be left out, and then the type of recording should be stated in the corresponding json file. I am somewhat hesitant to support your proposed change, because sometimes only the data after combining manual and autosamples is available (+ after additional modeling), and then it would be unclear what entity this data should have, as it is now a mixture between manual and autosamples. In this case, the entity could be left out (because it’s optional) and then described more thoroughly in the json. I know this is actually now derived data after combining the blood recordings, but I have seen cases where only this information is available. I would like not to block the sharing of this data. Let me know what you think of these potential concerns.

**Comment by bendhouseart** [comment]
The intent is for the `recording` entity to provide clarification if there are multiple blood recording files (or to simply provide more specificity). It looks like inverse is what needs to happen such that the rules yaml needs to be changed from `required` to `optional` [here](https://github.com/bids-standard/bids-specification/blob/5e2c32a90dc5130e19c546bfee5a1abe1324137e/src/schema/rules/datatypes/pet.yaml#L30).  

Would that be satisfactory? Or do you feel there still needs to be an update or additional example added to the [examples](https://github.com/bids-standard/bids-examples/tree/master/pet003/sub-01/ses-01/pet)?

**Comment by ghisvail** [comment]
The lack of consistency between the generated template in the docs, the wording of the paragraph and the example I linked to really makes the interpretation of this part of the spec quite confusing. Assuming the recording entity is indeed optional, then we should go with @bendhouseart suggestion to fix the rules generating the template and the inconsistent example I linked.

**Comment by mnoergaard** [comment]
The example you are referring to won't be inconsistent independent of whether the recording entity is required or optional. But after thinking more about this I am inclined to keep the recording entity required, as it provides clarity on the recording type, which is important for the person modelling the blood data and may not be present in the corresponding json file. My previous argument was indeed for derived blood data, so this is considerations to take into account for the PET derivatives.

**Review by mnoergaard — APPROVED** [review]

**Comment by mnoergaard** [comment]
Again, thanks for raising this issue @ghisvail.

**Comment by ghisvail** [comment]
You're welcome. Thank you all for taking time reviewing it and providing the additional pieces of context I was missing.

**Review by melanieganz — COMMENTED** [review]
Following the discussion above, I am fine with changing it to be required. The only question for me is what we do if people "only" have the merged (auto plus manual samples) left, e.g. for legacy data. Yes, this would in principle be a derivative, but I wonder if that's not in some cases the only raw data left. Do we anticipate this to become an issue?

**Comment by mathesong** [comment]
This feels like the 80/20 rule ought to take prevalence here: I think it's going to be a very small minority of PET datasets for which the auto / manual nature of the blood samples is no longer known.  I would support the change to make the recording type required.

From a purely pragmatic perspective, I've dealt with data where I've not had access to which samples were auto and manual, and one just has to treat everything as if it's manual data. I think that's quite an uncontroversial take (though of course you could do something fancy with guessing which are which, but it would be need to be tailored for the specifics of the dataset). So one strategy might be to just include in the text that if the auto/manual nature of the data is unknown, that everything can just be labelled as manual?

**Comment by ghisvail** [comment]
> So one strategy might be to just include in the text that if the auto/manual nature of the data is unknown, that everything can just be labelled as manual?

I'd be happy to add the proposed change to this PR. Is everyone in favour for it?

**Comment by mnoergaard** [comment]
Thanks for your take on this @mathesong - I agree with you. @ghisvail please feel free to add the proposed change to this PR. Thank you.

**Comment by ghisvail** [comment]
> Thanks for your take on this @mathesong - I agree with you. @ghisvail please feel free to add the proposed change to this PR. Thank you.

Done :+1:

File: src/04-modality-specific-files/09-positron-emission-tomography.md [hunk 1]
```diff
@@ -315,13 +315,12 @@ If collected, blood measurements of radioactivity are be stored in
...
 
-The OPTIONAL `recording` entity is used to distinguish sampling methods.
+The REQUIRED `recording` entity is used to distinguish sampling methods.
 For example, if an autosampler is used to record continuous blood samples,
 and manual measurements are also taken,
 then the files may have recording labels `autosampler` and `manual`,
 respectively.
-If multiple recording methods are used on the same PET acquisition,
-the `recording` entity MUST be used to distinguish them.
+if the sampling method is unknown, then recording is assumed `manual`.
```
Comment (tsalo) [inline src/04-modality-specific-files/09-positron-emission-tomography.md:323]:
```suggestion
If the sampling method is unknown, then recording SHOULD be set as `manual`.
```

Hopefully I'm interpreting this correctly. Saying that recording is "assumed `manual`" implies, to me at least, that the recording entity would be absent and the sampling method would be assumed to be manual, much like how MRI files without the `part` entity are assumed to reflect magnitude values.

**Review by tsalo — CHANGES_REQUESTED** [review]
I think the second edited sentence needs some tweaking. I do agree that this PR should be merged soon though.

**Review by tsalo — APPROVED** [review]
LGTM!

**Comment by sappelhoff** [comment]
Thanks everyone :+1: @mnoergaard could you briefly confirm that the validator does not need to be adjusted due to the changes here?

**Comment by mnoergaard** [comment]
LGTM - thanks @sappelhoff

**Pull request merged**